service.go

```diff
@@ -0,0 +1,337 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apiError "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Struct to build service watcher which looks after target services.
+type ServiceWatcher struct {
+	ctx       Context
+	svcFilter labels.Selector
+	informer  cache.SharedInformer
+	// Name space where to install service watcher,
```

Comment:
hmmm is this comment accurate? do we install the service watcher or mirror the services in there?

Reply:
Oops yep ! mirror it.